Fix docker elasticsearch for non-arm64

keithbauer opened the pull request:
## Description
<!--- Describe your changes -->
Manually run `/usr/local/bin/docker-entrypoint.sh` in the `command:` section in `docker-comopse.yml`. Also includes some `README.md` cleanup.

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Due to the differences between arm64 and amd64, the elasticsearch container needed to be handled differently. The arm64 builds uses `docker/Dockerfile-es654.arm64` to build the image while amd64 pulls the elasticsearch image from dockerhub. Due to this the arm64 Dockerfile can install elasticsearch plugins, but the amd64 version uses a command in `docker-compose.yml` to install the plugin. It seems like the entrypoint script wasn't getting ran so I've added it to the command. I'm not sure if it also needs to be added to `docker-compose.cicd.yml`.

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested on my windows machine and @oliviawongnyc tested on her non-m1 mac.

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

keithbauer committed:
Explicitly run

keithbauer committed:
Add elasticsearch volume to match docker-compose.arm64.yml

keithbauer committed:
Remove `make setup` command

The `setup` command was used to initialize the core submodule. Now that the submodule has been merged in the `setup` command was removed from the Makefile.

keithbauer committed:
Rename duplicate `openebooks-deploy-qa` to `openebooks-deploy-production`

mwbenowitz commented:
@keithbauer So this might be a bit of a left-field solution but I did some poking around and it turns out that bitnami publishes ES images that closely track the official ones but with some extra tooling around them: https://registry.hub.docker.com/r/bitnami/elasticsearch

In that is the ability to specify plugins you want install as env variables, which then get installed when the container starts up.

Since we're already pulling a bitnami image for minio, maybe this wouldn't be so bad? Just did a quick test and it seemed to solve the problem for me.

mwbenowitz committed:
BUGFIX Switch Container to bitnami/elasticsearch

This swaps out the Elastic.co container for one maintained by Bitnami. It includes some conveneince tooling that can install plugins, removing the need to define a custom command for the container. The bitnami maintained image closely tracks the officially supported images (up through 8.x at the time of this commit), so we do not foresee any issues with future support.

leonardr reviewed (approved):
Very happy with the simplification of moving to the bitnami image. I tested this on my AMD machine and am happy with the results.
